pkg/util/kubeversion/kubeversion.go

```diff
@@ -0,0 +1,75 @@
+package kubeversion
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	versionutil "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/discovery"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const fetchServerVersionInterval = time.Second * 10
+
+var (
+	KubeVersion1_27 = versionutil.MustParseSemantic("1.27.0")
+
+	serverVersion = &versionutil.Version{}
+	rwm           = &sync.RWMutex{}
+)
+
+type ServerVersionFetcher struct {
+	dc     discovery.DiscoveryInterface
+	ticker *time.Ticker
+}
+
+func NewServerVersionFetcher(dc discovery.DiscoveryInterface) *ServerVersionFetcher {
+	return &ServerVersionFetcher{
+		dc:     dc,
+		ticker: time.NewTicker(fetchServerVersionInterval),
+	}
+}
+
+// Start implements the Runnable interface to run ServerVersionFetcher as a controller.
+func (s *ServerVersionFetcher) Start(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx).WithName("serverVersionFetcher")
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	defer s.ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.V(5).Info("Context cancelled; stop fetching server version")
+			return nil
+		case <-s.ticker.C:
+			if err := s.FetchServerVersion(); err != nil {
+				log.Error(err, "Unable to fetch server version")
+			}
+			log.V(5).Info("Fetch server version", "serverVersion", serverVersion)
+		}
+	}
+}
+
+// FetchServerVersion gets API server version
+func (s *ServerVersionFetcher) FetchServerVersion() error {
+	clusterVersionInfo, err := s.dc.ServerVersion()
+	if err != nil {
+		return err
+	}
+	SetServerVersion(versionutil.MustParseSemantic(clusterVersionInfo.String()))
+	return nil
+}
+
+func SetServerVersion(v *versionutil.Version) {
+	rwm.Lock()
+	defer rwm.Unlock()
+	serverVersion = v
+}
+
+func LessThan(v *versionutil.Version) bool {
```

Comment:
It doesn't look like we are setting a fetcher in integration tests.
However, the webhook would still call this function. How does it work?
Is the assumption that empty (`0.0.0`) is LessThan `1.27.0`?